src/Modules/ContentBlocks/Domain/ContentBlock/Service/UpdateContentBlockWidget.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\Service;
+
+use ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\Command\CreateContentBlock;
+use ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\ContentBlock;
+use ForkCMS\Modules\ContentBlocks\Frontend\Widgets\Detail as DetailWidget;
+use ForkCMS\Modules\Frontend\Domain\Block\Block;
+use ForkCMS\Modules\Frontend\Domain\Block\BlockRepository;
+use ForkCMS\Modules\Frontend\Domain\Block\ModuleBlock;
+use ForkCMS\Modules\Internationalisation\Domain\Locale\Locale;
+use ForkCMS\Modules\Internationalisation\Domain\Translation\TranslationKey;
+
+class UpdateContentBlockWidget
+{
+    public function __construct(
+        private readonly BlockRepository $blockRepository
+    ) {
+    }
+
+    public function getNewExtraId(CreatecontentBlock $createcontentBlock): int
+    {
+        $block = new Block(
+            ModuleBlock::fromFQCN(DetailWidget::class),
+            TranslationKey::label('ContentBlocks'),
+            locale: $createcontentBlock->locale
+        );
+
+        $this->blockRepository->save($block);
+
+        return $block->getId();
+    }
+
+    public function updateWidget(ContentBlock $contentBlock): void
+    {
+        /** @var Block $block */
+        $block = $this->blockRepository->findOneById($contentBlock->getExtraId());
+
+        $block->getSettings()->add([
+            'extra_label' => $contentBlock->getTitle(),
+            'id' => $contentBlock->getId(),
+            'language' => $contentBlock->getLocale(),
```

Comment:
There is no need to put this here anymore since you can set the locale on a block

Reply:
The setting is still being set

Reply:
see https://github.com/forkcms/forkcms/pull/3555/commits/0eee4916220942e1607775ea492552d6a300601a